library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java

```diff
@@ -0,0 +1,168 @@
+
+    protected void populateBlocks()
+    {
+      long blockID = 1000;
+      for (int i = 0; i < FILE_CONTENTS.length; i++) {
+        int blockIndex = 1;
+        AbstractFileSplitter.FileMetadata fileMetadata = new AbstractFileSplitter.FileMetadata(outputPath);
+        fileMetadata.setRelativePath("file" + i);
+        int noOfBlocks = (int)((FILE_CONTENTS[i].length() / BLOCK_SIZE) + (((FILE_CONTENTS[i].length() % BLOCK_SIZE) == 0) ? 0 : 1));
+        long[] blockIds = new long[noOfBlocks];
+        for (int offset = 0; offset < FILE_CONTENTS[i].length(); offset += BLOCK_SIZE, blockID++) {
+          String blockContents;
+          if (offset + BLOCK_SIZE < FILE_CONTENTS[i].length()) {
+            blockContents = FILE_CONTENTS[i].substring(offset, offset + BLOCK_SIZE);
+          } else {
+            blockContents = FILE_CONTENTS[i].substring(offset);
+          }
+
+          AbstractBlockReader.ReaderRecord<Slice> readerRecord = new AbstractBlockReader.ReaderRecord<Slice>(blockID, new Slice(blockContents.getBytes()));
+          blockIds[blockIndex - 1] = blockID;
+          blockIndex++;
+          blockIdToExpectedContent.put(blockID, blockContents);
+          blockDataList.add(readerRecord);
+        }
+        fileMetadata.setBlockIds(blockIds);
+        fileMetadata.setNumberOfBlocks(noOfBlocks);
+        fileMetadatas.add(fileMetadata);
+      }
+    }
+
+    /* (non-Javadoc)
+     * @see org.junit.rules.TestWatcher#finished(org.junit.runner.Description)
+     */
+    @Override
+    protected void finished(Description description)
+    {
+      super.finished(description);
+
+      try {
+        FileUtils.deleteDirectory(new File(outputPath));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Rule
+  public PartFileWriterTest.TestMeta testMeta = new PartFileWriterTest.TestMeta();
+
+  @Test
+  public void testBlockWriting() throws IOException
+  {
+    Attribute.AttributeMap attributes = new Attribute.AttributeMap.DefaultAttributeMap();
+    attributes.put(DAG.DAGContext.APPLICATION_ID, "PartitionWriterTest");
+    attributes.put(DAG.DAGContext.APPLICATION_PATH, testMeta.outputPath);
+    testMeta.context = mockOperatorContext(1, attributes);
+
+    testMeta.underTest.setup(testMeta.context);
+    int index = 0;
+    for (int i = 0; i < FILE_CONTENTS.length; i++) {
+      testMeta.underTest.beginWindow(i);
+      testMeta.underTest.fileMetadataInput.process(testMeta.fileMetadatas.get(i));
+      for (int j = 0; j < testMeta.fileMetadatas.get(i).getNumberOfBlocks(); j++) {
+        testMeta.underTest.input.process(testMeta.blockDataList.get(index++));
+      }
+      testMeta.underTest.endWindow();
+    }
+    testMeta.underTest.committed(2);
+    testMeta.underTest.teardown();
+
+    File[] blockFileNames = testMeta.blocksDir.listFiles();
+    for (File blockFile : blockFileNames) {
+      int fileIndex = Integer.parseInt(blockFile.getName().split("\\.")[0].substring(4));
```

Comment:
If someone changes the number of blocks being tested then this substring may throw arrayIndexOutOfBounds. I think at the very least you should extract blockId to a class variable and do something like log10(blockId) if extracting the prefix in this way. A safer way would be to use a Regex to parse this out.